### PR TITLE
Add Dud

### PR DESCRIPTION
**Pull request by kevin-hanselman**
All of the below is taken from [Dud's README](https://github.com/kevin-hanselman/dud/blob/main/README.md)

## What is this tool for?

Dud is a lightweight tool for versioning data alongside source code and building data pipelines. In practice, Dud extends many of the benefits of source control to large binary data.

With Dud, you can commit, checkout, fetch, and push large files and directories with a simple command line interface. Dud stores recipes (a.k.a. stages) for retrieving your data in small YAML files. These stages can be stored in source control to link your data to your code. On top of that, stages can run the commands to generate the data, sort of like [Make](https://www.gnu.org/software/make/). Stages can be chained together to create data pipelines.

## What's the difference between this tool and similar ones?

Dud is heavily inspired by [DVC](https://dvc.org/). DVC addresses the need for data versioning and reproducibility, but its implementation is not without problems. My criticisms of DVC boil down to two things: speed and simplicity. By speed, I mean throughput and responsiveness. By simplicity, I mean doing less--both in project scope and amount of abstraction.

In terms of speed, Dud is generally much faster than DVC. In terms of simplicity, Dud has a smaller, more focused scope, and it is distributed as a single executable.

To summarize with an analogy: Dud is to DVC what [Flask](https://hackr.io/blog/flask-vs-django) is to [Django](https://hackr.io/blog/flask-vs-django). Both Dud and DVC have their strengths. If you want a "batteries included" suite of tools for managing machine learning projects, DVC may be a good fit for you. If data management is your main area of need and you want something lightweight and fast, Dud may be a better fit.

You can find a more thorough contrast to DVC in [Dud's README](https://github.com/kevin-hanselman/dud/blob/main/README.md#concrete-differences-with-dvc)

[1]: https://hackr.io/blog/flask-vs-django

---

Anyone who agrees with this pull request could submit an *Approve* review to it.

### CODE REVIEW

**Comment by kelvins** [comment]
Thanks @kevin-hanselman :rocket:
